resources-manager/include/ISystemResourceEntity.hpp

```diff
@@ -0,0 +1,29 @@
+// Concord
+//
+// Copyright (c) 2018-2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace concord::performance {
+class ISystemResourceEntity {
+ public:
+  virtual ~ISystemResourceEntity() = default;
+  // Implementation returns how much more pruned blocks can this entity support.
+  virtual int64_t getAvailableResources() const = 0;
+  // Measurements of handled operations. Can be transaction, prune blocks, etc.
```

Comment:
I think returning an opaque integer type is hard to reason about for the caller. Can we maybe define it it as "count of operations per some period" or similar?

Reply:
I thought that negative value could be used when a counter/metric exceeds it's capacity.

Reply:
Could you please explain that in a comment? Or maybe define another type that is even more explicit than an integer? It would make the interface easier to use.